src/variable.cpp

```diff
@@ -213,6 +215,19 @@ VariablePtr Variable::create(const std::string &name) noexcept
     return std::shared_ptr<Variable> {new Variable {name}};
 }
 
+bool Variable::doEqual(const EntityPtr &other) const
+{
+    if (NamedEntity::doEqual(other)) {
+        auto variable = std::dynamic_pointer_cast<libcellml::Variable>(other);
+        if (variable &&
+                this->initialValue() == variable->initialValue() &&
+                this->interfaceType() == variable->interfaceType()) {
+            return this->units() ? this->units()->equal(variable->units()) : true;
```

Comment:
this would return `true` is this variable has not units set even if the other variable has units, right?

Reply:
Yes, still more tests to be written to fully define what it means to be equal.